api/instructor/instructorMiddleware.js

```diff
@@ -14,6 +14,23 @@ const checkInstructorExist = async (req, res, next) => {
   }
 };
 
+// Retrieves the instructor id of the currently active account by way of the active account's profile id.
+// req.instructor_id is set to null if the active user is not an instructor
+// ../middleware/authRequired.js must be called before calling this middleware
+const getInstructorId = async (req, res, next) => {
+  const instructor_id = await Instructors.findInstructorIdByProfileId(
+    req.profile.profile_id
+  );
+  if (instructor_id) {
+    req.instructor_id = instructor_id.instructor_id;
+    next();
+  } else {
+    req.instructor_id = null;
+    next();
+  }
+};
+
 module.exports = {
   checkInstructorExist,
+  getInstructorId,
 };
```

Comment:
Nicely implemented middleware and model. looks good to go but just asking where it would be implemented in the instruct routers